### PR TITLE
Pipeline feature

### PR DESCRIPTION
**Pull request by ABostrom**
Some minor bug fixes to TSInstances

Pipeline feature for multivariate complex transformation and classification pipelines. 

Lots of toy examples showcased in the examples section

### CODE REVIEW

File: src/main/java/tsml/classifiers/multivariate/DimensionIndependentEnsemble.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ public String toString(){
             dist = modules[m].getClassifier().distributionForInstance(testInstance[m]);
             long predTime = System.currentTimeMillis() - startTime;
             
-            vs.storeModuleTestResult(modules[m], dist, predTime);
+            //vs.storeModuleTestResult(modules[m], dist, predTime);
```
Comment (James-Large) [inline src/main/java/tsml/classifiers/multivariate/DimensionIndependentEnsemble.java:153]: This intended? Cant remember the exact ramifications of it, but if was intentional and you've looked into it then that's all good

Comment (James-Large) [reply src/main/java/tsml/classifiers/multivariate/DimensionIndependentEnsemble.java:153]: Storing test results internally was mainly a helper thing for cawpe experiments iirc, not needed as generic functionality, so fine if it conflicts with something and isnt needed

**Comment by James-Large** [comment]
Looks solid, just see that comment

File: src/main/java/tsml/classifiers/multivariate/DimensionIndependentEnsemble.java (hunk 1 above)
Comment (ABostrom) [reply src/main/java/tsml/classifiers/multivariate/DimensionIndependentEnsemble.java:153]: Was getting a null exception when i was calling it.
Is what i think was happening is that the train was not being created properly. 
Can you review that class, as I think you understand the inner workings of the ensemble classes better.

Comment (James-Large) [reply src/main/java/tsml/classifiers/multivariate/DimensionIndependentEnsemble.java:153]: Likely the train results being null here then 
https://github.com/uea-machine-learning/tsml/blob/master/src/main/java/machine_learning/classifiers/ensembles/voting/ModuleVotingScheme.java#L120

Comment (James-Large) [reply src/main/java/tsml/classifiers/multivariate/DimensionIndependentEnsemble.java:153]: Ah ok, can look at this in more details tomorrow, but likely can just make DimIndEns extend EnhancedAbstractClassifier to get a trainresults object added on, and measure/store the build time naively as start time first line of buildClassifier, end time last line

May be other implications of making it an EAC too, would need another look

Comment (James-Large) [reply src/main/java/tsml/classifiers/multivariate/DimensionIndependentEnsemble.java]: Resolved, mainly via deletion of what is now a redundant class, really

**Comment by James-Large** [comment]
Comments resolved

**Pull request merged**